libkbfs/md_journal.go

```diff
@@ -464,6 +464,17 @@ func (j *mdJournal) convertToBranch(
 	j.j = tempJournal
 	j.branchID = bid
 
+	// Garbage-collect the old non-branch entries.  TODO: we'll
```

Comment:
I think we can do slightly better here, by cleaning up the branch entries if we encounter an error halfway. I think all you need is to keep track of the new md IDs, clean them up in the same defer that nukes the temp dir, and then set it to `allMdIDs` here.

I think one of the tests in `md_journal_test.go` has a case where the branch conversion fails, so if you add a GC check to those tests, we'll get coverage for this, too.